### DISCUSSION TITLE
Compiling OpenRADIOSS with MUMPS support in Apptainer and manually

### DISCUSSION THREAD

**Discussion by Irvise** [OpenRadioss/OpenRadioss#3177] [Q&A] [answered]
I recently saw the discussion thread _riginally posted by @JAWilliams123 in https://github.com/OpenRadioss/OpenRadioss/discussions/3149_, were some implicit buckling analysis cannot be performed with OpenRADIOSS as that feature (among a couple others) requires the use of MUMPS. I am assuming that MUMPS here refers to the [MUltifrontal Massively Parallel sparse direct Solver](https://mumps-solver.org/index.php?page=home) software. AFAIK, other open source software programs, such as [ElmerFEM](https://www.elmerfem.org/blog/), do not ship out precompiled versions of the software with MUMPS support either. This seems to be related with some license limitations from the CeCiLL-C license used by MUMPS.

ElmerFEM, however, supports MUMPS by allowing the users to compile their own version of ElmerFEM with MUMPS support on their own computers. From my IANAL (I Am Not A Lawyer) understanding, by forcing the users to compile the software themselves, the license limitations seem to be fulfilled without forcing any extra requirements on the original program. The CeCILL-C requirements and obligations are met without having to relicense the original software.

Therefore, my question... Can we/I potentially document in the manual how to compile/enable support of MUMPS (and therefore some extra features) in the OpenRADIOSS version? The Apptainer recipes could also be updated as their builds need to happen locally by the user.

I personally do not know how RADIOSS or [COMSOL](https://doc.comsol.com/5.5/doc/com.comsol.help.comsol/comsol_ref_solver.27.118.html) deals with this limitation (probably by having a commercial license), but maybe a similar approach to them could otherwise be used?

Best regards,
Fer

**Comment by laurent-altr** [comment, marked as answer]
Hello @Irvise , 
The issue is #3149 did not concern MUMPS directly. Even with MUMPS enabled, very few features ,such as buckling, are not available yet. The error message the user got is misleading.

By default, OpenRadioss is not activating the implicit par that depends on MUMPS. Nevertheless user's can still use it ([windows](https://github.com/OpenRadioss/OpenRadioss/blob/4d4e14c46fe5dda4d72596427c4efc4b63b170df/engine/build_windows.bat#L241), [linux](https://github.com/OpenRadioss/OpenRadioss/blob/4d4e14c46fe5dda4d72596427c4efc4b63b170df/engine/build_script.sh#L54))

**Reply by Irvise** [reply]
Hi Laurent,

thank you for your detailed answer. The message is indeed misleading :) I did not know that we could already build it with MUMPS support using directly supplied scripts from source. I will take a look at it.

Best regards,
Fer

**Reply by li2qn** [reply]
Hi there, it maybe a little while since this post. I am having the same issue to compile MUMPS for implicit solver. if you have done this with the windows build script above directly, could you please share how? I need some help here too. Thanks
